dhall/src/Dhall/Import.hs

```diff
@@ -746,6 +748,26 @@ toHeader _ = do
 getCacheFile
     :: (Alternative m, MonadIO m) => FilePath -> Dhall.Crypto.SHA256Digest -> m FilePath
 getCacheFile cacheName hash = do
+    cacheDirectory <- getOrCreateCacheDirectory cacheName
+
+    let cacheFile = cacheDirectory </> ("1220" <> show hash)
+
+    return cacheFile
+
+warnAboutMissingCache :: (Alternative m, MonadIO m) => m ()
+warnAboutMissingCache = alternative₀ <|> alternative₁
```

Comment:
You should probably warn inside the `get*` functions whenever something goes wrong so that the warning can be more specific about the reason for failure.  If you warn at the top level then it's hard to say for sure why the cache failed, but if you warn at the point of failure then you can narrow down the warning to a permissions problem or a specific environment variable being missing.

Reply:
Mmm, i had considered that alternative but then the warn will be emitted in each call to `getCacheFile` and afaiu it could be called several times in each dhall execution

Reply:
There is basically three cases: no env var is set, the pointed dir doesnt exist or it has not the required permission. I could try to handle all three but keeping a top level warn.